txscript/data/script_tests.json

```diff
@@ -1365,87 +1459,193 @@
+["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK", "NUMEQUALVERIFY must compare the first numeric stack item to the second numeric stack item, check if they are equal and verify the result"],
+["-1 -1", "NUMEQUALVERIFY 1", "NONE", "OK"],
+["1 2147483648", "NUMEQUALVERIFY 1", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUALVERIFY must fail with stack item >4 bytes"],
+
+["NUMNOTEQUAL test coverage"],
+["NOP", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items and NOP must not be treated as one"],
+["", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items, stack has no items"],
+["0", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items, stack has one item"],
+["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK", "NUMNOTEQUAL must compare the first numeric stack item to the second numeric stack item and check if they are not equal" ],
 ["111 10 1 ADD", "NUMNOTEQUAL", "NONE", "OK"],
-["11 10", "LESSTHAN NOT", "NONE", "OK"],
+["1 -1", "NUMNOTEQUAL", "NONE", "OK"],
+["1 2147483648", "NUMNOTEQUAL", "NONE", "ERR_OUT_OF_RANGE", "NUMNOTEQUAL must fail with stack item >4 bytes"],
+
+["LESSTHAN test coverage"],
+["NOP", "LESSTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHAN requires two items on the stack and NOP must not be treated as one"],
+["", "LESSTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHAN requires two items on the stack, stack has no items"],
+["0", "LESSTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHAN requires two items on the stack, stack has one item"],
+["11 10", "LESSTHAN NOT", "NONE", "OK", "LESSTHAN must check if the first numeric stack item is less than the second numeric stack item"],
 ["4 4", "LESSTHAN NOT", "NONE", "OK"],
 ["10 11", "LESSTHAN", "NONE", "OK"],
 ["-11 11", "LESSTHAN", "NONE", "OK"],
 ["-11 -10", "LESSTHAN", "NONE", "OK"],
-["11 10", "GREATERTHAN", "NONE", "OK"],
+["1 2147483648", "LESSTHAN", "NONE", "ERR_OUT_OF_RANGE", "LESSTHAN must fail with stack item >4 bytes"],
+
+["GREATERTHAN test coverage"],
+["NOP", "GREATERTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHAN requires two items on the stack and NOP must not be treated as one"],
+["", "GREATERTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHAN requires two items on the stack, stack has no items"],
+["0", "GREATERTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHAN requires two items on the stack, stack has one item"],
+["11 10", "GREATERTHAN", "NONE", "OK", "GREATERTHAN must check if the first numeric stack item is greater than the second numeric stack item"],
 ["4 4", "GREATERTHAN NOT", "NONE", "OK"],
 ["10 11", "GREATERTHAN NOT", "NONE", "OK"],
 ["-11 11", "GREATERTHAN NOT", "NONE", "OK"],
 ["-11 -10", "GREATERTHAN NOT", "NONE", "OK"],
-["11 10", "LESSTHANOREQUAL NOT", "NONE", "OK"],
+["1 2147483648", "GREATERTHAN", "NONE", "ERR_OUT_OF_RANGE", "GREATERTHAN must fail with stack item >4 bytes"],
```

Comment:
Same here regarding needing another test for the other operand being >4 bytes and comment.